compiler/optimizer/VPHandlers.cpp

```diff
@@ -2759,15 +2759,15 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
             }
          }
       }
-
+   int32_t frequency = vp->_curBlock->getGlobalNormalizedFrequency(vp->comp()->getFlowGraph());
    TR::VPConstraint * base = vp->getConstraint(node->getFirstChild(), isGlobal);
 
    if (base && base->getClass() && !vp->comp()->getOption(TR_DisableMarkingOfHotFields) &&
       node->getFirstChild()->getOpCode().hasSymbolReference() &&
       node->getFirstChild()->getSymbol()->isCollectedReference() &&
-      vp->_curBlock->getGlobalNormalizedFrequency(vp->comp()->getFlowGraph()) >= TR::Options::_hotFieldThreshold)
+      frequency >= TR::Options::_hotFieldThreshold)
       {
-      vp->comp()->fej9()->markHotField(vp->comp(), node->getSymbolReference(), base->getClass(), base->isFixedClass());
+      vp->comp()->fej9()->markHotField(frequency, vp->comp(), node->getSymbolReference(), base->getClass(), base->isFixedClass());
```

Comment:
markHotField is defined in Openj9, which creates OMR-on-OpenJ9 dependency (this code cannot be submitted without OpenJ9). There is also OpenJ9-on-OMR dependency (OpenJ9 GC helper that queries if Dynamic BF OMR flag is enabled), which together make the two PRs mutually dependent. 

Onc of OMR or OpenJ9 has to be split into two  PRs  (for the total of 3) to break the dependency in one of the directions, and then the PRs could be serially merged.

But, we'll worry about it once reviewers agree that the changes are semantically ok.